docs/source/deployment/index.rst

```diff
@@ -169,3 +169,144 @@ Almost all functionalities available in MLflow deployment can also be accessed v
+model dependencies and save them as part of the MLflow Model metadata. However, this inference is not always accurate and may
+miss some dependencies. This can cause errors when serving your model, such as "ModuleNotFoundError" or "ImportError". Here 
+is the steps to fix missing dependencies error.
+
+1. Check the missing dependencies
+*********************************
+The missing dependencies are listed in the error message. For example, if you see the following error message:
+
+.. code-block:: bash
+
+    ModuleNotFoundError: No module named 'cv2'
+
+2. Try adding the dependencies using the `predict` API
+******************************************************
+Now that you know the missing dependencies, you can create a new model version with the correct dependencies.
+However, creating a new model for trying new dependencies might be a bit tedious, particularly because you may need to
+iterate multiple times to find the correct solution. Instead, you can use the `predict` API to test your change without
+actually mutating the model.
+
+To do so, use the `pip-requirements-override` option to specify pip dependencies like `opencv-python==4.8.0`.
+
+.. tabs::
+
+    .. code-tab:: bash
+
+        mlflow models predict \
+            -m runs:/<run_id>/model \
+            -i <input_path> \
+            --pip-requirements-override opencv-python==4.8.0
+
+    .. code-tab:: python
+
+        import mlflow
+
+        mlflow.model.predict(
+            model_uri="runs:/<run_id>/model",
+            input_data=<input_data>,
+            pip_requirements="opencv-python==4.8.0",
+        )
+
+The specified dependencies will be installed to the virtual environment in addition to (or instead of) the dependencies
+defined in the model metadata. Since this doesn't mutate the model, you can iterate quickly and safely to find the correct dependencies.
+
+3. Update the model metadata
+****************************
+Once you find the correct dependencies, you can create a new model with the correct dependencies.
+To do so, specify either `pip_requirements` or `extra_pip_requirements` option when logging the model.
+
+.. code:: python
+
+    import mlflow
+
+    mlflow.pyfunc.log_model(
+        artifact_path="model",
+        python_model=python_model,
+        # If you want to define all dependencies from scratch, use `pip_requirements` option.
+        # Both options also accept a path to a pip requirements file e.g. requirements.txt.
+        extra_pip_requirements=["opencv-python==4.8.0"],
+    )
+
+.. note::
+
+    Alternatively, you can directly update the model metadata stored in the artifact storage. For example, `<path-to-model>/requirements.txt`
+    defines pip dependencies to be installed for serving your model (or `<path-to-model>/conda.yaml` if you use conda as a package manager).
+    However, this approach is not recommended because this is error-prone and also you need to do this for every model version.
```

Comment:
I don't think it's error prone if we give users good code to do this. Can we provide a snippet here? Maybe this deserves its own API (good to track with a follow-up JIRA).

Reply:
Ahh sorry here I mean manual file editting e.g. ssh&vi, let me rephrase it. Agree with having programatic way is good so will create a follow-up JIRA!